### PR TITLE
feat: add option to search user suggestions

### PR DESCRIPTION
**Pull request by rebelchris**
Option to search for user suggestions (based on name and username)

## Todo
- [x] Exclude banned users (awaiting Ole-Martin advice)

### CODE REVIEW

**Review by capJavert — APPROVED** [review]

**Comment by rebelchris** [comment]
@capJavert // @omBratteng added vodr exclude now.
Can you please check again?

**Review by capJavert — APPROVED** [review]

File: src/schema/search.ts [hunk 1]
```diff
@@ -388,6 +410,37 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
...
+      source,
+      { query, limit }: SearchSuggestionArgs,
+      ctx,
+    ): Promise<GQLSearchSuggestionsResults> => {
+      const searchQuery = ctx.con
+        .createQueryBuilder()
+        .select(`id, name as title, username as subtitle, image`)
+        .from('user', 'u')
+        .where('u.infoConfirmed = TRUE')
+        .andWhere("u.id != '404'")
```
Comment (omBratteng) [inline src/schema/search.ts:428]: Instead of hardcoding `404`, lets use the `ghostUser.id` variable?

File: src/schema/search.ts [hunk 2]
```diff
@@ -388,6 +410,37 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
...
+            return qb
+              .where(`name ILIKE :query`, {
+                query: `%${query}%`,
+              })
+              .orWhere(`username ILIKE :query`, {
+                query: `%${query}%`,
+              });
+          }),
+        )
+        .andWhere(whereVordrFilter('u'))
```
Comment (omBratteng) [inline src/schema/search.ts:440]: Yup, looks good. Should return `COALESCE((u.flags ->> 'vordr')::boolean, false) = false`

File: __tests__/search.ts [hunk 3]
```diff
@@ -528,3 +529,121 @@ describe('query searchSourceSuggestions', () => {
...
+
+  it('should only return vodr false users', async () => {
+    await con.getRepository(User).update(
+      { id: '3' },
+      {
+        flags: updateFlagsStatement<User>({
+          vordr: true,
+        }),
+      },
+    );
```
Comment (omBratteng) [inline __tests__/search.ts:615]: Could use the user `vordr` from `badUsersFixture`?

**Review by omBratteng — APPROVED** [review]
Looks good, minor comments

File: src/schema/search.ts (hunk 1 above)
Comment (rebelchris) [reply src/schema/search.ts:428]: Good point, probably should fix it in some other places as well!

File: __tests__/search.ts (hunk 3 above)
Comment (rebelchris) [reply __tests__/search.ts:615]: Ah think this is easier as we can then test again the succesfull test above if that's ok?

**Pull request merged**